crates/flux-middle/src/global_env.rs

```diff
@@ -206,8 +206,13 @@ impl<'genv, 'tcx> GlobalEnv<'genv, 'tcx> {
 
     pub fn impl_trait_ref(
         self,
-        impl_id: DefId,
+        impl_id: LocalDefId,
```

Comment:
This should also work for external impls so we must take a `DefId`. I'd create a method in `GlobalEnv` like so

```rust
fn extern_id_of(self, def_id: DefId) -> Option<DefId> {
    self.map().extern_id_of(def_id.as_local()?)
}
```

Then the code below changes to

```rust
let impl_id = self.extern_id_of(impl_id).unwrap_or(impl_id);
```